variables.tf

```diff
@@ -1,12 +1,24 @@
 # ------------------------------------------------------------------------------
 # Variables
 # ------------------------------------------------------------------------------
+
 variable "name_prefix" {
   description = "A prefix used for naming resources."
 }
 
+variable "lifecycle_count_number" {
```

Comment:
Since this is not a variable in a terraform resource, we get to name it ourselves - but I think it would be more clear if it was `max_image_count` or `max_images_retained`?